sklearn/utils/estimator_checks.py

```diff
@@ -3050,6 +3055,120 @@ def check_class_weight_balanced_linear_classifier(name, Classifier):
+        sample_weight_zero_weight_first = np.where(y == 0, 0, sample_weight)
+        estimator_sw = clone(estimator_orig)
+        set_random_state(estimator_sw, random_state=0)
+        estimator_sw.fit(X, y, sample_weight=sample_weight_zero_weight_first)
+
+        # Checking if the output is the same for multiple outputs
+        for method in ["predict", "predict_proba", "decision_function", "transform"]:
+            if hasattr(estimator_orig, method):
+                pred_cw = getattr(estimator_cw, method)(X)
+                pred_sw = getattr(estimator_sw, method)(X)
+                pred_exclude = getattr(estimator_sw, method)(X)
+                assert_allclose_dense_sparse(pred_cw, pred_sw, err_msg=err_msg_sw)
+                assert_allclose_dense_sparse(
+                    pred_cw, pred_exclude, err_msg=err_msg_exclude
+                )
+
+
+@ignore_warnings(category=FutureWarning)
+def check_interaction_of_class_and_sample_weight_excluding_samples(
+    name, estimator_orig
+):
+    """Setting sample_weight to 0 is equivalent to removing corresponding
+    samples even when using non uniform class_weight.
+    """
+    # Note: this test is similar to check_sample_weights_invariance and to
+    # check_interaction_of_class_and_sample_weight_excluding_class.
+    rng = np.random.RandomState(42)
+    err_msg = (
+        f"For {name}, while using class weight, setting some sample's weight "
+        "to zero is not equivalent to excluding those samples from training."
+    )
+
+    tags = _safe_tags(estimator_orig)
+    max_classes = 2 if tags["binary_only"] else 9
+
+    for n_classes in range(2, max_classes + 1):
+        X, y = make_classification(
+            n_samples=200,
+            n_classes=n_classes,
+            n_informative=2 * n_classes,
+            random_state=0,
+        )
+
+        sample_weight = rng.binomial(1, 0.8, size=y.shape[0])
+        class_weight_dict = {
+            cls: np.random.RandomState(cls).randint(1, 6) for cls in range(n_classes)
+        }
+
+        # Model with some sample_weight set to 0
+        estimator_sw = clone(estimator_orig).set_params(class_weight=class_weight_dict)
+        set_random_state(estimator_sw, random_state=0)
+        estimator_sw.fit(X, y, sample_weight=sample_weight)
+
+        # Model excluding samples with 0 sample weight
+        X_exclude = X[sample_weight > 0]
+        y_exclude = y[sample_weight > 0]
+        estimator_exclude = clone(estimator_orig).set_params(
+            class_weight=class_weight_dict
+        )
+        set_random_state(estimator_exclude, random_state=0)
+        estimator_exclude.fit(X_exclude, y_exclude)
+
+        # Checking if the output is the same for multiple outputs
+        for method in ["predict", "predict_proba", "decision_function", "transform"]:
+            if hasattr(estimator_orig, method):
+                pred_sw = getattr(estimator_sw, method)(X)
+                pred_exclude = getattr(estimator_sw, method)(X)
```

Comment:
```suggestion
                pred_exclude = getattr(estimator_exclude, method)(X)
```

Reply:
Let's pretend this didn't happen 🫠 hahahah